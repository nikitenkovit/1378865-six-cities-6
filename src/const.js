export const OfferType = {
  APARTMENT: `Apartment`,
  ROOM: `Private Room`,
  HOUSE: `House`,
  HOTEL: `Hotel`
};

export const PathValue = {
  MAIN_SCREEN: `/`,
  FAVORITES_SCREEN: `/favorites`,
  ROOM_SCREEN: `/offer/:id`
};

export const CityCoordinate = {
  AMSTERDAM: {
    latitude: 52.38333,
    longitude: 4.9,
    zoom: 12
  }
}; // скорее всего временно

export const OffersListClassName = {
  CITY_PLACES: `cities__places-list`,
  NEAR_PLACES: `near-places__list`
};

export const BookmarkButtonProperty = {
  ROOM_SCREEN: {
    buttonClass: `property__bookmark-button`,
    iconClass: `property__bookmark-icon`,
    iconWidth: `31`,
    iconHeight: `33`
  },
  OFFER_CARD: {
    buttonClass: `place-card__bookmark-button`,
    iconClass: `place-card__bookmark-icon`,
    iconWidth: `18`,
    iconHeight: `19`
  }
};
