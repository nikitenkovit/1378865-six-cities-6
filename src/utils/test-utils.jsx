import React from "react";
import {Router} from 'react-router-dom';
import {render as pureRender} from "@testing-library/react";
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";
import browserHistory from "../history";

const render = (
    ui, {
      store = {},
      ...otherOptions
    }) => {
  const wrapper = ({children}) => (
    <Provider store={configureStore()(store)}>
      <Router history={browserHistory}>
        {children}
      </Router>
    </Provider>
  );
  return pureRender(ui, {wrapper, ...otherOptions});
};

// Реэкспортим библиотеку, чтобы не нужно было ее подключать.
export * from '@testing-library/react';
export {render};
