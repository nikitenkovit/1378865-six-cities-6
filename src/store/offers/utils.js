import {SortingType} from "../../const";

export const sortingFunction = (offers, type) => (a, b) => {
  switch (type) {
    case SortingType.HIGH_TO_LOW:
      return b.price - a.price;
    case SortingType.LOW_TO_HIGH:
      return a.price - b.price;
    case SortingType.TOP_RATED:
      return b.rating - a.rating;
  }
  return offers;
};
