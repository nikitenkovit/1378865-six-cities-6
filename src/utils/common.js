import offersData from '../mocks/offersData';
import {SortingType} from "../const";

export const getRatingStarValue = (number) => {
  return Math.round(number) * 20 + `%`;
};

export const adaptOfferData = (data) => ({
  bedrooms: data.bedrooms,
  city: data.city,
  description: data.description,
  goods: data.goods,
  host: {
    avatarUrl: data.host.avatar_url,
    id: data.host.id,
    isPro: data.host.is_pro,
    name: data.host.name,
  },
  id: data.id,
  images: data.images,
  isFavorite: data.is_favorite,
  isPremium: data.is_premium,
  location: data.location,
  maxAdults: data.max_adults,
  previewImage: data.preview_image,
  price: data.price,
  rating: data.rating,
  title: data.title,
  type: data.type,
});

export const adaptCommentsData = (data) => ({
  comment: data.comment,
  date: data.date,
  id: data.id,
  rating: data.rating,
  user: {
    avatarUrl: data.user.avatar_url,
    id: data.user.id,
    isPro: data.user.is_pro,
    name: data.user.name,
  }
});

export const getOffers = () => offersData.map(adaptOfferData); // временно. пока нет api

export const getOffersByCity = (state) => {
  return getOffers().filter((offer) => offer.city.name === state.CITY.name);
};

export const sortingFunction = (offers, type) => (a, b) => {
  switch (type) {
    case SortingType.HIGH_TO_LOW:
      return b.price - a.price;
    case SortingType.LOW_TO_HIGH:
      return a.price - b.price;
    case SortingType.TOP_RATED:
      return b.rating - a.rating;
  }
  return offers;
};
