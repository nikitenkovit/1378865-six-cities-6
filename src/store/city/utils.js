import {DEFAULT_CURRENT_CITY} from "../../const";
import {getOffers} from "../../utils/common";

export const getCurrentCity = (state) => {
  if (state && state.CITY) {
    return state.CITY;
  }
  return getCities()
    .find((city) => city.name === DEFAULT_CURRENT_CITY);
};

export const getCities = () => {
  const cities = getOffers()
    .reduce((generalOffer, offer) => {
      if (!generalOffer.hasOwnProperty(offer.city.name)) {
        generalOffer[offer.city.name] = {...offer.city};
      }

      return generalOffer;
    }, {});

  return Object.values(cities);
};
