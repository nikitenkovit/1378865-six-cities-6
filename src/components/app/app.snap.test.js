import React from 'react';
import {render} from "../../utils/test-utils";
import App from './app';
import {AppRoute, LoadStatus, AuthorizationStatus, DefaultCitiesList} from "../../const";
import * as redux from 'react-redux';

import browserHistory from "../../history";

const mockOffer = {
  "bedrooms": 3,
  "city": {
    "location": {
      "latitude": 52.5909553943508,
      "longitude": 4.95309666406198,
      "zoom": 12
    },
    "name": `Amsterdam`
  },
  "description": `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
  "goods": [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
  "host": {
    "avatarUrl": `test`,
    "id": 3,
    "isPro": true,
    "name": `test`
  },
  "id": 1,
  "images": [`./img/apartment-01.jpg`, `./img/apartment-02.jpg`, `./img/apartment-03.jpg`, `./img/room.jpg`, `./img/apartment-02.jpg`, `./img/apartment-03.jpg`],
  "isFavorite": true,
  "isPremium": false,
  "location": {
    "latitude": 52.5909553943508,
    "longitude": 4.95309666406198,
    "zoom": 8
  },
  "maxAdults": 4,
  "previewImage": `./img/apartment-01.jpg`,
  "price": 120,
  "rating": 4.8,
  "title": `Beautiful & luxurious studio at great location`,
  "type": `apartment`
};

const mockStore = {
  OFFERS: {
    items: [mockOffer],
    status: LoadStatus.SUCCESS
  },
  USER: {
    authorizationStatus: AuthorizationStatus.NO_AUTH,
    user: null
  },
  CITIES: {
    items: DefaultCitiesList,
    current: {
      name: `Paris`,
      location: {
        latitude: 48.85661,
        longitude: 2.351499,
        zoom: 13
      }
    }
  },
  OFFER_LOCATION: {
    latitude: 48.83861,
    longitude: 2.350499,
  },
  CURRENT_OFFER: {
    current: mockOffer,
    reviews: [{
      "comment": `test`,
      "date": `2019-05-08T14:13:56.569Z`,
      "id": 1,
      "rating": 4,
      "user": {
        "avatarUrl": `test`,
        "id": 4,
        "isPro": false,
        "name": `test`
      }
    }],
    nearest: [mockOffer],
    status: LoadStatus.SUCCESS
  },
  FAVORITES: {
    items: [],
    status: LoadStatus.SUCCESS
  }
};

describe(`Test routing`, () => {
  jest.spyOn(redux, `useDispatch`);

  it(`Render 'MainScreen' when user navigate to ${AppRoute.MAIN} url`, () => {

    const {container} = render(
        <App/>,
        {store: mockStore}
    );

    expect(container).toMatchSnapshot();
  });

  it(`Render 'SignInScreen' when user navigate to ${AppRoute.LOGIN} url`, () => {

    browserHistory.push(AppRoute.LOGIN);

    const {container} = render(
        <App/>,
        {store: mockStore}
    );

    expect(container).toMatchSnapshot();
  });

  it(`Render 'RoomScreen' when user navigate to ${AppRoute.ROOM} url`, () => {
    browserHistory.push(AppRoute.ROOM);

    jest.spyOn(redux, `useDispatch`);

    const {container} = render(
        <App/>,
        {store: mockStore});
    expect(container).toMatchSnapshot();
  });
});
