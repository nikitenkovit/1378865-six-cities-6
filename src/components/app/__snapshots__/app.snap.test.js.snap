// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test routing Render 'MainScreen' when user navigate to / url 1`] = `
<div>
  <div
    class="page page--gray page--main"
  >
    <header
      class="header"
    >
      <div
        class="container"
      >
        <div
          class="header__wrapper"
        >
          <div
            class="header__left"
          >
            <a
              class="header__logo-link header__logo-link--active"
              href="/"
            >
              <img
                alt="6 cities logo"
                class="header__logo"
                height="41"
                src="img/logo.svg"
                width="81"
              />
            </a>
          </div>
          <nav
            class="header__nav"
          >
            <ul
              class="header__nav-list"
              style="flex-direction: column; align-items: flex-end;"
            >
              <li
                class="header__nav-item user"
              >
                <a
                  class="header__nav-link header__nav-link--profile"
                  href="/favorites"
                >
                  <span
                    class="header__user-name user__name"
                  >
                    Sign in
                  </span>
                </a>
              </li>
            </ul>
          </nav>
        </div>
      </div>
    </header>
    <main
      class="page__main page__main--index"
    >
      <h1
        class="visually-hidden"
      >
        Cities
      </h1>
      <div
        class="tabs"
      >
        <section
          class="locations container"
        >
          <ul
            class="locations__list tabs__list"
          >
            <li
              class="locations__item"
            >
              <a
                class="locations__item-link tabs__item tabs__item--active"
                href="#"
              >
                <span>
                  Paris
                </span>
              </a>
            </li>
            <li
              class="locations__item"
            >
              <a
                class="locations__item-link tabs__item false"
                href="#"
              >
                <span>
                  Cologne
                </span>
              </a>
            </li>
            <li
              class="locations__item"
            >
              <a
                class="locations__item-link tabs__item false"
                href="#"
              >
                <span>
                  Brussels
                </span>
              </a>
            </li>
            <li
              class="locations__item"
            >
              <a
                class="locations__item-link tabs__item false"
                href="#"
              >
                <span>
                  Amsterdam
                </span>
              </a>
            </li>
            <li
              class="locations__item"
            >
              <a
                class="locations__item-link tabs__item false"
                href="#"
              >
                <span>
                  Hamburg
                </span>
              </a>
            </li>
            <li
              class="locations__item"
            >
              <a
                class="locations__item-link tabs__item false"
                href="#"
              >
                <span>
                  Dusseldorf
                </span>
              </a>
            </li>
          </ul>
        </section>
      </div>
      <div
        class="cities"
      >
        <div
          class="cities__places-container container"
        >
          <section
            class="cities__places places"
          >
            <h2
              class="visually-hidden"
            >
              Places
            </h2>
            <b
              class="places__found"
            >
              0
               places to stay in 
              Paris
            </b>
            <form
              action="#"
              class="places__sorting"
              method="get"
            >
              <span
                class="places__sorting-caption"
              >
                Sort by 
              </span>
              <span
                class="places__sorting-type"
                tabindex="0"
              >
                Popular
                <svg
                  class="places__sorting-arrow"
                  height="4"
                  width="7"
                >
                  <use
                    xlink:href="#icon-arrow-select"
                  />
                </svg>
              </span>
              <ul
                class="places__options places__options--custom
    null"
              >
                <li
                  class="places__option \\" tabIndex=\\"0\\"
    places__option--active"
                >
                  Popular
                </li>
                <li
                  class="places__option \\" tabIndex=\\"0\\"
    null"
                >
                  Price: low to high
                </li>
                <li
                  class="places__option \\" tabIndex=\\"0\\"
    null"
                >
                  Price: high to low
                </li>
                <li
                  class="places__option \\" tabIndex=\\"0\\"
    null"
                >
                  Top rated first
                </li>
              </ul>
            </form>
            <div
              class="cities__places-list places__list
     tabs__content"
            />
          </section>
          <div
            class="cities__right-section"
          >
            <section
              class="cities__map map"
            >
              <div
                class="leaflet-container leaflet-grab leaflet-touch-drag"
                id="map"
                style="height: 100%; position: relative;"
                tabindex="0"
              >
                <div
                  class="leaflet-pane leaflet-map-pane"
                  style="left: 0px; top: 0px;"
                >
                  <div
                    class="leaflet-pane leaflet-tile-pane"
                  >
                    <div
                      class="leaflet-layer "
                      style="z-index: 1;"
                    >
                      <div
                        class="leaflet-tile-container leaflet-zoom-animated"
                        style="z-index: 18; left: 0px; top: 0px;"
                      >
                        <img
                          alt=""
                          class="leaflet-tile"
                          role="presentation"
                          src="https://b.basemaps.cartocdn.com/rastertiles/voyager/13/4149/2818.png"
                          style="width: 256px; height: 256px; left: -130px; top: -72px;"
                        />
                      </div>
                    </div>
                  </div>
                  <div
                    class="leaflet-pane leaflet-shadow-pane"
                  />
                  <div
                    class="leaflet-pane leaflet-overlay-pane"
                  />
                  <div
                    class="leaflet-pane leaflet-marker-pane"
                  />
                  <div
                    class="leaflet-pane leaflet-tooltip-pane"
                  />
                  <div
                    class="leaflet-pane leaflet-popup-pane"
                  />
                </div>
                <div
                  class="leaflet-control-container"
                >
                  <div
                    class="leaflet-top leaflet-left"
                  />
                  <div
                    class="leaflet-top leaflet-right"
                  />
                  <div
                    class="leaflet-bottom leaflet-left"
                  />
                  <div
                    class="leaflet-bottom leaflet-right"
                  >
                    <div
                      class="leaflet-control-attribution leaflet-control"
                    >
                      <a
                        href="https://leafletjs.com"
                        title="A JS library for interactive maps"
                      >
                        Leaflet
                      </a>
                       | © 
                      <a
                        href="https://www.openstreetmap.org/copyright"
                      >
                        OpenStreetMap
                      </a>
                       contributors © 
                      <a
                        href="https://carto.com/attributions"
                      >
                        CARTO
                      </a>
                    </div>
                  </div>
                </div>
              </div>
            </section>
          </div>
        </div>
      </div>
    </main>
  </div>
</div>
`;

exports[`Test routing Render 'RoomScreen' when user navigate to /offer/:id url 1`] = `
<div>
  <div
    class="page"
  >
    <header
      class="header"
    >
      <div
        class="container"
      >
        <div
          class="header__wrapper"
        >
          <div
            class="header__left"
          >
            <a
              class="header__logo-link header__logo-link--active"
              href="/"
            >
              <img
                alt="6 cities logo"
                class="header__logo"
                height="41"
                src="img/logo.svg"
                width="81"
              />
            </a>
          </div>
          <nav
            class="header__nav"
          >
            <ul
              class="header__nav-list"
              style="flex-direction: column; align-items: flex-end;"
            >
              <li
                class="header__nav-item user"
              >
                <a
                  class="header__nav-link header__nav-link--profile"
                  href="/favorites"
                >
                  <span
                    class="header__user-name user__name"
                  >
                    Sign in
                  </span>
                </a>
              </li>
            </ul>
          </nav>
        </div>
      </div>
    </header>
    <main
      class="page__main page__main--property"
    >
      <section
        class="property"
      >
        <div
          class="property__gallery-container container"
        >
          <div
            class="property__gallery"
          >
            <div
              class="property__image-wrapper"
            >
              <img
                alt="Photo studio"
                class="property__image"
                src="./img/apartment-01.jpg"
              />
            </div>
            <div
              class="property__image-wrapper"
            >
              <img
                alt="Photo studio"
                class="property__image"
                src="./img/apartment-02.jpg"
              />
            </div>
            <div
              class="property__image-wrapper"
            >
              <img
                alt="Photo studio"
                class="property__image"
                src="./img/apartment-03.jpg"
              />
            </div>
            <div
              class="property__image-wrapper"
            >
              <img
                alt="Photo studio"
                class="property__image"
                src="./img/room.jpg"
              />
            </div>
            <div
              class="property__image-wrapper"
            >
              <img
                alt="Photo studio"
                class="property__image"
                src="./img/apartment-02.jpg"
              />
            </div>
            <div
              class="property__image-wrapper"
            >
              <img
                alt="Photo studio"
                class="property__image"
                src="./img/apartment-03.jpg"
              />
            </div>
          </div>
        </div>
        <div
          class="property__container container"
        >
          <div
            class="property__wrapper"
          >
            <div
              class="property__name-wrapper"
            >
              <h1
                class="property__name"
              >
                Beautiful & luxurious studio at great location
              </h1>
              <button
                class="property__bookmark-button property__bookmark-button--active button"
                type="button"
              >
                <svg
                  class="property__bookmark-icon"
                  height="33"
                  width="31"
                >
                  <use
                    xlink:href="#icon-bookmark"
                  />
                </svg>
                <span
                  class="visually-hidden"
                >
                  To bookmarks
                </span>
              </button>
            </div>
            <div
              class="property__rating rating"
            >
              <div
                class="property__stars rating__stars"
              >
                <span
                  style="width: 100%;"
                />
                <span
                  class="visually-hidden"
                >
                  Rating
                </span>
              </div>
              <span
                class="property__rating-value rating__value"
              >
                4.8
              </span>
            </div>
            <ul
              class="property__features"
            >
              <li
                class="property__feature property__feature--entire"
              >
                Apartment
              </li>
              <li
                class="property__feature property__feature--bedrooms"
              >
                3 Bedrooms
              </li>
              <li
                class="property__feature property__feature--adults"
              >
                Max 
                4
                 adults
              </li>
            </ul>
            <div
              class="property__price"
            >
              <b
                class="property__price-value"
              >
                €
                120
              </b>
              <span
                class="property__price-text"
              >
                 night
              </span>
            </div>
            <div
              class="property__inside"
            >
              <h2
                class="property__inside-title"
              >
                What's inside
              </h2>
              <ul
                class="property__inside-list"
              >
                <li
                  class="property__inside-item"
                >
                  Heating
                </li>
                <li
                  class="property__inside-item"
                >
                  Kitchen
                </li>
                <li
                  class="property__inside-item"
                >
                  Cable TV
                </li>
                <li
                  class="property__inside-item"
                >
                  Washing machine
                </li>
                <li
                  class="property__inside-item"
                >
                  Coffee machine
                </li>
                <li
                  class="property__inside-item"
                >
                  Dishwasher
                </li>
              </ul>
            </div>
            <div
              class="property__host"
            >
              <h2
                class="property__host-title"
              >
                Meet the host
              </h2>
              <div
                class="property__host-user user"
              >
                <div
                  class="property__avatar-wrapper user__avatar-wrapper property__avatar-wrapper--pro"
                >
                  <img
                    alt="Host avatar"
                    class="property__avatar user__avatar"
                    height="74"
                    src="test"
                    width="74"
                  />
                </div>
                <span
                  class="property__user-name"
                >
                  test
                </span>
              </div>
              <div
                class="property__description"
              >
                <p
                  class="property__text"
                >
                  A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.
                </p>
              </div>
            </div>
            <section
              class="property__reviews reviews"
            >
              <h2
                class="reviews__title"
              >
                Reviews · 
                <span
                  class="reviews__amount"
                >
                  1
                </span>
              </h2>
              <ul
                class="reviews__list"
              >
                <li
                  class="reviews__item"
                >
                  <div
                    class="reviews__user user"
                  >
                    <div
                      class="reviews__avatar-wrapper user__avatar-wrapper"
                    >
                      <img
                        alt="Reviews avatar"
                        class="reviews__avatar user__avatar"
                        height="54"
                        src="test"
                        width="54"
                      />
                    </div>
                    <span
                      class="reviews__user-name"
                    >
                      test
                    </span>
                  </div>
                  <div
                    class="reviews__info"
                  >
                    <div
                      class="reviews__rating rating"
                    >
                      <div
                        class="reviews__stars rating__stars"
                      >
                        <span
                          style="width: 80%;"
                        />
                        <span
                          class="visually-hidden"
                        >
                          Rating
                        </span>
                      </div>
                    </div>
                    <p
                      class="reviews__text"
                    >
                      test
                    </p>
                    <time
                      class="reviews__time"
                      datetime="2019-05-08T14:13:56.569Z"
                    >
                      May 2019
                    </time>
                  </div>
                </li>
              </ul>
            </section>
          </div>
        </div>
        <section
          class="property__map map"
        >
          <div
            class="leaflet-container leaflet-grab leaflet-touch-drag"
            id="map"
            style="height: 100%; position: relative;"
            tabindex="0"
          >
            <div
              class="leaflet-pane leaflet-map-pane"
              style="left: 0px; top: 0px;"
            >
              <div
                class="leaflet-pane leaflet-tile-pane"
              >
                <div
                  class="leaflet-layer "
                  style="z-index: 1;"
                >
                  <div
                    class="leaflet-tile-container leaflet-zoom-animated"
                    style="z-index: 18; left: 0px; top: 0px;"
                  >
                    <img
                      alt=""
                      class="leaflet-tile"
                      role="presentation"
                      src="https://a.basemaps.cartocdn.com/rastertiles/voyager/13/4208/2683.png"
                      style="width: 256px; height: 256px; left: -182px; top: -244px;"
                    />
                  </div>
                </div>
              </div>
              <div
                class="leaflet-pane leaflet-shadow-pane"
              />
              <div
                class="leaflet-pane leaflet-overlay-pane"
              />
              <div
                class="leaflet-pane leaflet-marker-pane"
              >
                <img
                  alt=""
                  class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
                  src="./img/pin.svg"
                  style="margin-left: -15px; margin-top: -15px; width: 30px; height: 30px; left: 0px; top: 0px; z-index: 0; z-index: 0;"
                  tabindex="0"
                />
                <img
                  alt=""
                  class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
                  src="./img/pin-active.svg"
                  style="margin-left: -15px; margin-top: -15px; width: 30px; height: 30px; left: 0px; top: 0px; z-index: 0;"
                  tabindex="0"
                />
              </div>
              <div
                class="leaflet-pane leaflet-tooltip-pane"
              />
              <div
                class="leaflet-pane leaflet-popup-pane"
              />
            </div>
            <div
              class="leaflet-control-container"
            >
              <div
                class="leaflet-top leaflet-left"
              />
              <div
                class="leaflet-top leaflet-right"
              />
              <div
                class="leaflet-bottom leaflet-left"
              />
              <div
                class="leaflet-bottom leaflet-right"
              >
                <div
                  class="leaflet-control-attribution leaflet-control"
                >
                  <a
                    href="https://leafletjs.com"
                    title="A JS library for interactive maps"
                  >
                    Leaflet
                  </a>
                   | © 
                  <a
                    href="https://www.openstreetmap.org/copyright"
                  >
                    OpenStreetMap
                  </a>
                   contributors © 
                  <a
                    href="https://carto.com/attributions"
                  >
                    CARTO
                  </a>
                </div>
              </div>
            </div>
          </div>
        </section>
      </section>
      <div
        class="container"
      >
        <section
          class="near-places places"
        >
          <h2
            class="near-places__title"
          >
            Other places in the neighbourhood
          </h2>
          <div
            class="near-places__list places__list
     "
          >
            <article
              class="near-places__card place-card"
            >
              <div
                class="near-places__image-wrapper place-card__image-wrapper"
              >
                <a
                  href="/offer/1"
                >
                  <img
                    alt="Place image"
                    class="place-card__image"
                    height="200"
                    src="./img/apartment-01.jpg"
                    width="260"
                  />
                </a>
              </div>
              <div
                class=" place-card__info"
              >
                <div
                  class="place-card__price-wrapper"
                >
                  <div
                    class="place-card__price"
                  >
                    <b
                      class="place-card__price-value"
                    >
                      €
                      120
                    </b>
                    <span
                      class="place-card__price-text"
                    >
                      / night
                    </span>
                  </div>
                  <button
                    class="place-card__bookmark-button place-card__bookmark-button--active button"
                    type="button"
                  >
                    <svg
                      class="place-card__bookmark-icon"
                      height="19"
                      width="18"
                    >
                      <use
                        xlink:href="#icon-bookmark"
                      />
                    </svg>
                    <span
                      class="visually-hidden"
                    >
                      To bookmarks
                    </span>
                  </button>
                </div>
                <div
                  class="place-card__rating rating"
                >
                  <div
                    class="place-card__stars rating__stars"
                  >
                    <span
                      style="width: 100%;"
                    />
                    <span
                      class="visually-hidden"
                    >
                      Rating
                    </span>
                  </div>
                </div>
                <h2
                  class="place-card__name"
                >
                  <a
                    href="/offer/1"
                  >
                    Beautiful & luxurious studio at great location
                  </a>
                </h2>
                <p
                  class="place-card__type"
                >
                  Apartment
                </p>
              </div>
            </article>
          </div>
        </section>
      </div>
    </main>
  </div>
</div>
`;

exports[`Test routing Render 'SignInScreen' when user navigate to /login url 1`] = `
<div>
  <div
    class="page page--gray page--login"
  >
    <header
      class="header"
    >
      <div
        class="container"
      >
        <div
          class="header__wrapper"
        >
          <div
            class="header__left"
          >
            <a
              class="header__logo-link header__logo-link--active"
              href="/"
            >
              <img
                alt="6 cities logo"
                class="header__logo"
                height="41"
                src="img/logo.svg"
                width="81"
              />
            </a>
          </div>
          <nav
            class="header__nav"
          >
            <ul
              class="header__nav-list"
              style="flex-direction: column; align-items: flex-end;"
            >
              <li
                class="header__nav-item user"
              >
                <a
                  class="header__nav-link header__nav-link--profile"
                  href="/favorites"
                >
                  <span
                    class="header__user-name user__name"
                  >
                    Sign in
                  </span>
                </a>
              </li>
            </ul>
          </nav>
        </div>
      </div>
    </header>
    <main
      class="page__main page__main--login"
    >
      <div
        class="page__login-container container"
      >
        <section
          class="login"
        >
          <h1
            class="login__title"
          >
            Sign in
          </h1>
          <form
            action="#"
            class="login__form form"
            method="post"
          >
            <div
              class="login__input-wrapper form__input-wrapper"
            >
              <label
                class="visually-hidden"
              >
                E-mail
              </label>
              <input
                class="login__input form__input"
                name="email"
                placeholder="Email"
                type="email"
              />
            </div>
            <div
              class="login__input-wrapper form__input-wrapper"
            >
              <label
                class="visually-hidden"
              >
                Password
              </label>
              <input
                class="login__input form__input"
                name="password"
                placeholder="Password"
                type="password"
              />
            </div>
            <button
              class="login__submit form__submit button"
              type="submit"
            >
              Sign in
            </button>
          </form>
        </section>
        <section
          class="locations locations--login locations--current"
        >
          <div
            class="locations__item"
          >
            <a
              class="locations__item-link"
              href="/"
            >
              <span>
                Amsterdam
              </span>
            </a>
          </div>
        </section>
      </div>
    </main>
  </div>
</div>
`;
